apis/logging/v1/input_source_types.go

```diff
@@ -0,0 +1,52 @@
+package v1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+)
+
+// NOTE: The Enum validation on SourceSpec.Type must be updated if the list of types changes.
+
+// SourceSpec is a union of input source types.
+//
+// The fields of this struct define the set of known source types.
+type SourceSpec struct {
+	HTTP *HTTPSource `json:"http,omitempty"`
+}
+
+const (
+	FormatK8S  = "k8s"  // Log events in k8s list format, e.g. API audit log events.
+	FormatText = "text" // Newline-separated plain-text log records.
+)
+
+// HTTPSource receives encoded logs as a HTTP endpoint.
+type HTTPSource struct {
+
+	// FIXME TLS server configuration - can this be handled in the Service,
+	// or do we need vector to have certificates? Can the service be bypassed?
+
+	// FIXME What to do about service name clashes? Can't generate random name, service name is used to locate the source.
+
+	// Name of the source.
+	// A Service resource is generated for each source,
+	// the name must be a unique Service name in the ClusterLogForwarder's namespace.
+	//
+	// +required
+	Name string `json:"name"`
```

Comment:
Why  do we need a "name" if the inputspec already has one that can be defined by the user.  This looks to be redundant

Reply:
I was wondering myself. @alanconway, do we really need another Name here?

Reply:
Perhaps not, we need a name for the generated service but the name from the inputSpec should be enough.
I will review the API and see if we can simplify by making the source-input relationship 1-1.